internal/route/node_filter.go

```diff
@@ -107,6 +107,32 @@ func (f *IsDoneSyncing) Apply(_ metadata.RequestMetadata, upstreamConfig *config
 	return true
 }
 
+type IsLatencyAcceptable struct {
+	healthCheckManager checks.HealthCheckManager
+	logger             *zap.Logger
+}
+
+func (f *IsLatencyAcceptable) Apply(requestMetadata metadata.RequestMetadata, upstreamConfig *config.UpstreamConfig, _ int) bool {
+	upstreamStatus := f.healthCheckManager.GetUpstreamStatus(upstreamConfig.ID)
+
+	latencyCheck, _ := upstreamStatus.LatencyCheck.(*checks.LatencyCheck)
+
+	if latencyCheck.ShouldRun {
+		// TODO(polsar): If unhealthy, set the delta by which the check failed.
+		//  For example, if the configured rate is 0.25 and the current error rate is 0.27,
+		//  the delta is 0.27 - 0.25 = 0.02. This value can be used to rank upstreams by the
+		//  degree to which they are unhealthy. This would help us choose the upstream to
+		//  route to if all upstreams are unhealthy AND `alwaysRoute` config option is true.
+		upstreamConfig.HealthStatus = latencyCheck.GetUnhealthyReason(requestMetadata.Methods)
+	}
+
+	// TODO(polsar): Note that for LatencyCheck only, we always return true. The health status
+	//  of the upstream is instead contained in the struct's HealthStatus field. This is a bit
+	//  clunky. Eventually, we want to change the signature of the Apply method. This will require
+	//  significant refactoring.
+	return true
```

Comment:
Wait, I might be missing something. Don't we want to check the circuit breaker? (I haven't looked at your last PR yet).

Reply:
Left a [comment](https://github.com/satsuma-xyz/node-gateway/pull/133#discussion_r1746183119) in the other PR that could affect this